spec/integration/outputs/delete_spec.rb

```diff
@@ -0,0 +1,61 @@
+require_relative "../../../spec/es_spec_helper"
+require "logstash/outputs/elasticsearch"
+
+
+describe "Versioned delete", :integration => true, :version_greater_than_equal_to_2x => true do
+  require "logstash/outputs/elasticsearch"
+
+  def get_es_output( options={} )
+    settings = {
+      "manage_template" => true,
+      "index" => "logstash-delete",
+      "template_overwrite" => true,
+      "hosts" => get_host_port(),
+    }
+    LogStash::Outputs::ElasticSearch.new(settings.merge!(options))
+  end
+
+  before :each do
+    @es = get_client
+    # Delete all templates first.
+    # Clean ES of data before we start.
+    @es.indices.delete_template(:name => "*")
+    # This can fail if there are no indexes, ignore failure.
+    @es.indices.delete(:index => "*") rescue nil
+    @es.indices.refresh
+  end
+
+  context "when delete only" do
+    it "should ignore non-monotonic external_version updates" do
+      id = "ev2"
+      subject = get_es_output({ 'document_id' => id, "external_version" => "99", "action" => "index" })
+      subject.register
```

Comment:
Can you rejigger these specs to use `before` blocks a bit better? Can you use the `subject {}` declaration in rspec itself with a separate `let(:options) { {'document_id' => id, ...}` declaration? It's a lot cleaner than chucking it all in the `it` block.